chia/wallet/wallet_node.py

```diff
@@ -603,20 +604,23 @@ async def receive_state_from_peer(
 
         all_tasks: List[asyncio.Task] = []
         target_concurrent_tasks: int = 20
-        num_concurrent_tasks: int = 0
+        concurrent_tasks_cs_heights: List[uint32] = []
+
+        # Ensure the list is sorted
+        items.sort(key=last_change_height_cs)
 
         async def receive_and_validate(inner_states: List[CoinState], inner_idx_start: int):
+            nonlocal concurrent_tasks_cs_heights
             try:
                 assert self.validation_semaphore is not None
                 async with self.validation_semaphore:
-                    assert self.wallet_state_manager is not None
                     if header_hash is not None:
                         assert height is not None
                         for inner_state in inner_states:
                             self.add_state_to_race_cache(header_hash, height, inner_state)
                             self.log.info(f"Added to race cache: {height}, {inner_state}")
                     if trusted:
-                        valid_states = inner_states
+                        valid_states = inner_states.copy()
```

Comment:
I don't think this is needed anymore, it was before because we were mutating things inside `new_coin_state`